### PR TITLE
Support cross-building

### PR DESCRIPTION
**Pull request by axw**
- Specify --platform in the golang build stage
- Set GOOS and GOARCH env vars
- Remove apt-get upgrades; changed base image to chainguard/static

I don't think we need buildx?

### CODE REVIEW

**Review by lahsivjar — APPROVED** [review]
LGTM!

File: Containerfile [hunk 1]
```diff
@@ -37,18 +42,6 @@ LABEL \
...
-RUN \
-  apt-get update > /dev/null \
-  && apt-get upgrade -y \
-  && apt-get install -y --no-install-recommends \
-    "apt-utils=*" \
-    "ca-certificates=*" \
-    "curl=*" \
-    "gnupg=*" \
-  && apt-get clean \
-  && rm -rf /var/lib/apt/lists/*
```
Comment (inge4pres) [inline Containerfile:50]: Removing this will break the tilt/AWS setup that relies on `curl` to generate the API key.

**Review by inge4pres — CHANGES_REQUESTED** [review]
Sorry to block,it would break the load testing setup, see comment

File: Containerfile (hunk 1 above)
Comment (axw) [reply Containerfile:50]: ugh :)

Thanks for catching that, didn't realise/forgot we were doing this.

**Comment by axw** [comment]
@inge4pres thanks for catching that, should be resolved with https://github.com/elastic/apm-managed-service/pull/760. PTAL.

**Review by inge4pres — APPROVED** [review]
Thanks, tested it out with all other recent changes and it works ™️ 
Nice addition 👍🏼

**Pull request merged**